sympy/stats/tests/test_stochastic_process.py

```diff
@@ -194,3 +194,14 @@ def test_BernoulliProcess():
     assert P(B[5] > 0, B[5]) == BernoulliDistribution(0.6, 0, 1)
     raises(ValueError, lambda: P(3))
     raises(ValueError, lambda: P(B[3] > 0, 3))
+
+    # test issue 19456
+    expr = Sum(B[t], (t, 0, 4))
+    expr2 = Sum(B[t], (t, 1, 3))
+    expr3 = Sum(B[t]**2, (t, 1, 3))
+    assert expr.doit() == B[0] + B[1] + B[2] + B[3] + B[4]
+    assert expr2.doit() == Y
+    assert expr3.doit() == B[1]**2 + B[2]**2 + B[3]**2
+    assert B[2*t].free_symbols == {B[2*t], t}
+    assert B[4].free_symbols == {B[4]}
+    assert B[x*t].free_symbols == {B[x*t], x, t}
```

Comment:
When doing it on `IndexedBase`, I get:
```python
In [4]: (A[i]).free_symbols
Out[4]: {A, i, A[i]}

In [5]: (A[i]).expr_free_symbols
Out[5]: {A[i]}
```

Maybe this should return `{B[x*t], x, t, B}` ? Did you also test `.expr_free_symbols`? `.expr_free_symbols` is like `.free_symbols` except it stops as soon as the expression tree is not an expression/math formula anymore.

Reply:
> Maybe this should return `{B[x*t], x, t, B}`

IMHO, `B` should not be included in free symbols, as it is not an `Expr` object. Also, `expr_free_symbols` works as expected:
```python
>>> B[x*t].expr_free_symbols
{B[t*x]}
>>> B[4].expr_free_symbols
{B[4]}
>>> B[2*t].expr_free_symbols
{B[2*t]}
```

Reply:
ping @Upabjojr

Reply:
How would it affect the fix if we return `B` as well?

Reply:
```python
>>> Eq(B[1], 0).as_set()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/smit/Smitlunagariya/sympy/sympy/logic/boolalg.py", line 165, in as_set
    raise NotImplementedError("Sorry, as_set has not yet been"
NotImplementedError: Sorry, as_set has not yet been implemented for multivariate expressions
```

Reply:
Okay, so may be we can exclude `B` from the results.

Reply:
Maybe `Indexed` should be modified not to return its base element in its free symbols... but that's another issue.